backend/src/routes/auth_routes.py

```
@@ -0,0 +1,85 @@
+    db=SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+#NOTE:signup me pydantic model hi use kar rahe jsime json data accpet kar rahe par login me stricly form-data le rahe jo standard hai in like with google github me bhi kaam kar jaye isliye
+    # login ke liye bhi pydnatic model define karke json data le sakte hai ye method jyada compatible hai      
+
+#since yaha pe db call lagegi db ka session chaiye to use it as middleware
+@authRouter.post("/signup",response_model=schema.UserOutput) #ye response_model will check ki jo fields UserOutput me diye hai bas vahi jaye baki sensitive info like password na jaye aur ye json ke tarah bhi usko parse kar deta hai
+def signup(user:schema.UserCreateSignup,db:Session=Depends(get_db)): # get_db database ka sesssion return karega of type Session jo ki db me assign ho jayega
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)